addon/authenticators/oauth2-password-grant.js

```diff
@@ -129,17 +129,17 @@ export default BaseAuthenticator.extend({
 
   /**
     When authentication fails, the rejection callback is provided with the whole
-    XHR object instead of it's response JSON or text.
+    fetch response object instead of it's response JSON or text.
 
     This is useful for cases when the backend provides additional context not
     available in the response body.
 
-    @property rejectWithXhr
+    @property rejectWithResponse
     @type Boolean
     @default false
     @public
   */
-  rejectWithXhr: false,
+  rejectWithResponse: false,
```

Comment:
This cannot be renamed as that would be a breaking change. Instead `rejectWithXhr` should be a deprecated alias for `rejectWithResponse`.